Update Article.md (#1)
K-S-K/CV: Articles/05_EWReliability/Article.md

[:point_left: Back to the main page](../../README.md)

**Development period:** August 2012 - May 2017.

> **Practical application:** To use in production and in the CI cycle[^1].

Reviewer: ```suggestion
**Practical application:** To measure the availability factor of software in production and in the CI cycle[^1].
```

**Project purpose:** The program complex reliability parameters and status analysis.

**Project description:** 
Initially we had one point of interest: to measure availability factor of our software (a cell operator billing system) to be sure it meets the requirements of our customer (cell operator). They asked us to provide an availability factor = 0.99995.
We've researched methodology for measuring a time period we can qualify as "working interval" and a time period we can call "not working interval".

> The ratio of the working interval to the whole observation time is the availability factor we need. 

Reviewer: ```suggestion
The ratio of a calculated sum of the working interval to the whole observation time is the availability factor we need. 
```

**My part in this project:** 
- Theory research;
- RND (different experiments with collecting and analysis of the measures);
- The project architecture development;
- Creating of the initial version of testing software deployment subsystem;

> - Creating several beta traffic models;

Author: Если тут имелось в виду не бета, поправь пожалуйста, я так поняла)

> - Creating several beta traffic models;

Reviewer: ```suggestion
- Creating two initial traffic models (Voice and SMS);
```

- Creating a cell commutator emulator (with traffic generator schedulers and response time and content analyzers);
- Creating a state reporting component for the software components which must be measured;
- Creating a scanner component to collect data from the observed applications and a scanner itself to collect the data to the SQL Server;
- Creating the first version of data analysis implementation;

> - Creating a desktop software tool to view the reliability information in different representations;

Reviewer: ```suggestion
- Creating a desktop software tool to view the reliability information in different representations (table and diagram);
```

> - Creating a desktop tool to control the work of the reliability data collecting system remotely over a cluster (start and stop, selection of a version of measuring software to be working on the node, configuration of data collecting in every mode);

Reviewer: ```suggestion
- Creating a desktop tool to control the work of the reliability data collecting system remotely over a cluster (start and stop, selection of a version of measuring software to be working on the node, configuration of data collecting on every node);
```

- Sharing the work between several members of our team.

**Evolving features of the project we implemented with wonderfull people I worked with:** 
- Configurable deployment over the cluster subsystem;
- Many different traffic models for the load testing subsystem;
- Suspendable queue of measures to survive the database server unavailability periods without keeping all measures in memory (sometimes it was extremely necessary); 
- Several additional measuring subsystems like SQLServer availability observation component and MongoDB Health monitoring component, because SQLServer and MongoDB don't naturally provide data we need to observe them;
- Reliability Interval Marking on DB subsystem which is running on SQL Server by the SQL Server Agent and makes ready-to-use intervals of work for every observing software instance;
- Tree-like table interval representation for the UI and for the reports;

> - Reliability diagram - a scalable graphical representation of the reliability intervals on a single time scale for manual visual analysis of emergency situations evolution process;

Reviewer: ```suggestion
- Reliability diagram - a scalable graphical representation of the reliability intervals on a common time scale for manual visual analysis of emergency situations evolution process;
```
